File: app/src/main/java/com/sysu/edu/academic/LeaveReturnRegistrationActivity.java
package com.sysu.edu.academic;

import static com.sysu.edu.api.CommonUtil.toStringOrDefault;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.alibaba.fastjson2.JSONArray;
import com.alibaba.fastjson2.JSONObject;
import com.sysu.edu.databinding.ActivityLeaveReturnRegistrationBinding;
import com.sysu.edu.model.XgxtModel;

import java.util.ArrayList;

public class LeaveReturnRegistrationActivity extends AppCompatActivity {
    
    XgxtModel model;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityLeaveReturnRegistrationBinding binding = ActivityLeaveReturnRegistrationBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        model = new XgxtModel(this);
        binding.toolbar.setNavigationOnClickListener(_ -> supportFinishAfterTransition());
        LeaveReturnRegistrationViewModel viewModel = new ViewModelProvider(this).get(LeaveReturnRegistrationViewModel.class);
        model.getMessage().observe(this, message -> {
            JSONObject response = message.getSecond();
            if (response != null && response.getInteger("code") == 200) {
                if (message.getFirst() == 0) {
                    JSONArray data;
                    if ((data = response.getJSONArray("data")) != null && !data.isEmpty()) {
                        ArrayList<String> years = new ArrayList<>();
                        data.forEach(o -> years.add(toStringOrDefault(((JSONObject) o).getString("label"), "")));
                        binding.years.setSimpleItems(years.toArray(new String[0]));
                        binding.years.setOnItemClickListener((_, _, position, _) -> viewModel.year.setValue(data.getJSONObject(position).getString("value")));
                        binding.years.setText(years.get(0), false);
                        viewModel.year.setValue(data.getJSONObject(0).getString("value"));
                    }
                }
            }
        });
        getYears();
    }
    
    void getYears() {
        model.addAndNext("jjrlfx/api/sm-jjrlfx/student/school-year", 0);
    }
}

File: app/src/main/java/com/sysu/edu/academic/LeaveReturnRegistrationActivity.kt
package com.sysu.edu.academic

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.alibaba.fastjson2.JSONObject
import com.sysu.edu.BaseActivity
import com.sysu.edu.api.CommonUtil
import com.sysu.edu.databinding.ActivityLeaveReturnRegistrationBinding
import com.sysu.edu.model.XgxtModel

class LeaveReturnRegistrationActivity : BaseActivity() {
	lateinit var model: XgxtModel
	override fun onDestroy() {
		super.onDestroy()
		model.dispose()
	}
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		model = XgxtModel(this)
		val viewModel = ViewModelProvider(this).get<LeaveReturnRegistrationViewModel>(LeaveReturnRegistrationViewModel::class.java)
		val binding = ActivityLeaveReturnRegistrationBinding.inflate(layoutInflater).apply {
			toolbar.setNavigationOnClickListener { supportFinishAfterTransition() }
		}
		setContentView(binding.root)
		model.message.observe(this, Observer { message: CommonUtil.Tuple2<Int, JSONObject> ->
			val response = message.second
			if (response.getInteger("code") == 200) {
				if (message.first == 0) {
					response.getJSONArray("data")?.let {
						val years = ArrayList<String?>()
						it.forEach { o: Any? -> years.add((o as JSONObject).getString("label", "")) }
						binding.years.apply {
							setSimpleItems(years.toTypedArray<String?>())
							setOnItemClickListener { _: AdapterView<*>?, _: View?, position: Int, _: Long ->
								viewModel.year.value = it.getJSONObject(position).getString("value")
							}
							setText(years[0], false)
						}
						viewModel.year.value = it.getJSONObject(0).getString("value")
					}
				}
			}
		})
		years
	}
	
	val years: Unit
		get() {
			model.addAndNext("jjrlfx/api/sm-jjrlfx/student/school-year", 0)
		}
}